lua/sim/weapon.lua

```diff
@@ -472,29 +473,43 @@ Weapon = ClassWeapon(WeaponMethods, DebugWeaponComponent) {
     OnMotionHorzEventChange = function(self, new, old)
     end,
 
+    damageTableCache = nil,
+    -- Determine the validity of the damageTableCache:
+    -- nil -> never initialized;
+    -- false -> stale, needs to be updated;
+    -- true -> fresh, ready to use
+    damageTableCacheValid = nil,
     ---@param self Weapon
-    GetDamageTableInternal = function(self)
+    ---@return WeaponDamageTable
+    RefreshDamageTable = function(self)
         local weaponBlueprint = self.Blueprint
         ---@type WeaponDamageTable
-        local damageTable = {}
+        local damageTable
+
+        -- Setup the table for values that won't change later
+        if self.damageTableCacheValid == nil then
```

Comment:
I don't think keeping state through nil and false is great.
```suggestion
        if not damageTable then
```